docs/messaging/messaging-spans.md

```diff
@@ -376,92 +424,83 @@ All attributes that are specific for a messaging system SHOULD be populated in `
 
 ### Topic with multiple consumers
 
-Given is a process P, that publishes a message to a topic T on messaging system MS, and two processes CA and CB, which both receive the message and process it.
-
-```
-Process P:  | Span Prod1 |
---
-Process CA:              | Span CA1 |
---
-Process CB:                 | Span CB1 |
+Given is a publisher that publishes a message to a queue "T" on RabbitMQ, and two consumers which both receive the message.
```

Comment:
The text being replaced describes a scenario where a publisher is publishing to a topic T, but this is replaced with publishing to a queue T. I think publishing to a topic here makes much more sense than publishing to a queue. A queue is a point-to-point communication mechanism whereas a topic is intended to reach multiple consumers. I'll concede it's usually technically possible for multiple consumers to receive a message from a queue, but this is only in corner cases where, perhaps, a consumer receives the message and is then disconnected before it is able to acknowledge the message. I think it would be better to illustrate multiple consumers with a normal scenario rather than a corner case scenario.

There may be a more definitive source of information, but this reference shows how publishing to a RabbitMQ topic would be received by multiple consumers:
https://www.rabbitmq.com/tutorials/tutorial-five-python.html

Reply:
You're right, things got mixed up here. Also, the RabbitMQ example had the title "Topic with multiple consumers", which is confusing.

I re-wrote the examples as follows:
* "Topic with multiple consumers": Kafka, topic, push-based consumption (deliver).
* "Batch receiving": RabbitMQ, queue, pull-based consumption (receive).

Reply:
I'm not sure RabbitMQ is the best messaging system to use to illustrate batch message processing. I'm not a RabbitMQ expert, but in a quick search I wasn't able to see how batch messaging could be performed with "out of the box" RabbitMQ. I did find batching examples where RabbitMQ is used under the hood , but in all cases I found, this is where the batching was performed by some software on top of the underlying RabbitMQ messaging system (ex. Spring Cloud Streams, Spring Boot, and various home grown layers or application workarounds).

In the [RabbitMQ docs](https://www.rabbitmq.com/documentation.html) and one of the client guides ([Java Client](https://www.rabbitmq.com/api-guide.html)), I wasn't able to find any explanation how to send or receive with batches.

I think it may be preferable to continue using RabbitMQ as the example publishing on topic T to multiple consumers, and continue to use Kafka as the batch consumer example from topic T.

Reply:
Ok, I ended up with those two scenarios now:

* "Topic with multiple consumers": RabbitMQ, topic exchange, push-based consumption (deliver).
* "Batch receiving": Kafka, topic, pull-based consumption (receive).

Reply:
This looks good to me, thanks @pyohannes.